xarray_extras/csv.py

```diff
@@ -16,7 +17,7 @@
 __all__ = ("to_csv",)
 
 
-def to_csv(x: xarray.DataArray, path: str, *, nogil: bool = True, **kwargs):
+def to_csv(x: xarray.DataArray, path: Path, *, nogil: bool = True, **kwargs):
```

Comment:
I'm happy to accept a PR that adds support for Path.
It's not OK to remove support for str. Could you amend the PR to support both?

Reply:
Of course, it is not clear to me on-sight why the string would be needed though. Will try to fix.

Reply:
so that a user can write `to_csv(x, "somename.csv")`

Reply:
Using a Path makes up for type clarity, will fail if the input string cannot be converted to a proper path, is cross-platform and more. The user can always write `to_csv(x, "somename.csv")` and `x` we can convert to a string internally if need be, simply via `str(x)`. I updated all occurrences of `path` to be `Path` and converted to string via `str(path)` where (I think) it is necessary to get a string.  Will add to the existing PR or create a new if necessary. Unless I miss some specific requirement.

Reply:
Indeed, I missed to test exactly what you asked for before!  I added now `str | Path` and a check and attempt to convert to Path internally, in case user input is a `str`ing. I hope this works out nicely now.